src/crystal/system/panic.cr

```diff
@@ -0,0 +1,19 @@
+module Crystal::System
+  # Prints a system error message on the standard error then exits.
+  #
+  # Prefer `RuntimeError.from_os_error` unless you can't allocate any memory
+  # (e.g. stop the world).
+  def self.panic(syscall_name : String, error = nil)
+    error ||=
+      {% if flag?(:unix) %}
+        Errno.value
+      {% elsif flag?(:win32) %}
+        WinError.value
```

Comment:
In this case wouldn't `#message` allocate memory?

Reply:
Damn. Both cases (Errno#message and WinError#message) allocate since they call `String.new` and `String.from_utf16` :facepalm: